File: 53. Maximum Subarray/solution.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        dp = []
        max_sum = nums[0]
        for i in range(1, nums):
            dp.append(max(dp[i-1]+nums[i], nums[i]))
            max_sum = max(max_sum, dp[i])
        return max_sum

File: 53. Maximum Subarray/solution.go
package main

// Given an integer array nums, find the subarray with the largest sum, and return its sum.

func maxSubArray(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
    globalMax := nums[0]
	currentMax := nums[0]
	for i := 1; i < len(nums); i++ {
		currentMax = max(nums[i], currentMax+nums[i])
		globalMax = max(globalMax, currentMax)
	}
	return globalMax
}